scripts/linux/nested-edge-deploy-agent.sh

```diff
@@ -6,7 +6,7 @@ function create_certificates() {
     echo "Installing test root certificate bundle."
 
     echo "Generating edge device certificate"
-    device_name=$(curl ipecho.net/plain)
+    device_name=$(ip route get 8.8.8.8 | sed -n '/src/{s/.*src *\([^ ]*\).*/\1/p;q}')
```

Comment:
https://github.com/Azure/azure-cli/issues/2851

Reply:
It is slower and less reliable. Will stay on DNS server which is better.